vernac/vernacentries.ml

```diff
@@ -2719,16 +2721,36 @@ let interp ?(verbosely=true) ~st cmd =
     Vernacstate.invalidate_cache ();
     iraise exn
 
-let interp_qed_delayed_proof ~proof ~info ~st ?loc pe : Vernacstate.t =
+(* Interpreting a possibly delayed proof *)
+let interp_qed_delayed ~proof ~info ~st pe : Vernacstate.t =
   let stack = st.Vernacstate.lemmas in
   let stack = Option.cata (fun stack -> snd @@ Vernacstate.LemmaStack.pop stack) None stack in
+  let () = match pe with
+    | Admitted ->
+      save_lemma_admitted_delayed ~proof ~info
+    | Proved (_,idopt) ->
+      save_lemma_proved_delayed ~proof ~info ~idopt in
+  { st with Vernacstate.lemmas = stack }
+
+let rec interp_qed_delayed_control ~proof ~info ~st ?loc ~control pe : Vernacstate.t =
```

Comment:
This code seems a to be a duplication of iterp_control above.
It should be factored and abstracted over the base case.

Reply:
I tried but that's pretty tricky. I'll give it another go.

Reply:
Gave it a go, it doesn't look too bad.